src/lib389/cli/dscontainer

```diff
@@ -425,6 +427,19 @@ def begin_healthcheck(ds_proc, log_exception):
     return (True, False)
 
 
+def stop():
+    pid=int(open(PID_FILE).read())
+    parent = psutil.Process(pid)
+    procs = parent.children(recursive=True)
+    procs.append(parent)
+    for p in procs:
+        p.terminate()
```

Comment:
If you terminate the parent process, all child processes would be terminated too. Is there a reason to go after each child process separately?

Reply:
I think each process is supposed to cleanly shutdown all child processes, but if they don't then they become orphans (parent goes away first) or zombies (parent signals child, but does not wait for exit status) and in both cases they are re-parented to PID 1, which is supposed to do the clean up for you so in many cases people don't notice.   This may be one of those sneaky complications with UNIX, though I might be over-thinking this.   In our case dscontainer may or may not be PID 1 and dscontainer may or may not be handling cleanup properly (hard to tell).

Reply:
Also worth mentioning that the stop code is complicated by timing and checking as it first ask the child politely to stop (SIGTERM), then waits a while, then if child didn't exit gracefully in specified time then issues SIGKILL.   I think this whole thing is rather complicated if you do it gracefully.  Which is why having a built-in stop is valuable so users don't have to think about all these details or hack together a less robust kill command as did originally.

Reply:
See my comment above https://github.com/389ds/389-ds-base/pull/6062#discussion_r1474810237